daprblog/content/posts/2023/dapr_flomesh/index.md

```diff
@@ -0,0 +1,546 @@
+
+The Dapr toolkit provides rich features such as service invocation, resilience policies, state storage, publish/subscribe, binding, distributed locking, name resolution, etc. However, it does not support advanced service governance functions such as canary release and cross-cluster service invocation.
+
+### Flomesh Service Mesh
+
+After the rise of microservices architecture, as the scale grows larger, the difficulty and fragmentation of service governance have significantly increased. The emergence of service meshes solves these problems. A service mesh is a dedicated infrastructure layer for handling communication between services. Through it, functionality such as observability, traffic management, and security can be added transparently without adding it to your code.
+
+[Flomesh](https://flomesh.io) service mesh uses the programmable proxy [Pipy](https://github.com/flomesh-io/pipy) as the core to provide east-west and north-south traffic management. Through L7-based traffic management capabilities, it breaks through the network isolation between computing environments, proposes a virtual flat network, and enables communication between applications in different computing environments. Flomesh service mesh can be imagined as a "big mesh" that covers multiple clusters.
+
+{{< imgproc flomesh_dapr_blog_arch.png  Resize "800x" />}}
+
+## Demo 
+
+The image below shows an example of how Dapr and Flomesh service mesh can be integrated to enable cross-cluster service communication. The example consists of two components: a server-side NodeApp and a client-side curl.
+
+The server-side NodeApp is a Dapr application, which is modified based on the NodeApp in the [Dapr hello-kubernetes](https://github.com/dapr/quickstarts/tree/master/tutorials/hello-kubernetes) example. When it returns a response, it displays the name of the current cluster.
+
+The client-side curl is used to send requests to NodeApp but is not declared as a Dapr application. This demonstrates how Flomesh service mesh can enable communication between a Dapr application and a non-Dapr application, enabling cross-cluster communication.
+
+
+{{< imgproc flomesh_dapr_blog_demo_arch.png  Resize "1200x" />}}
+
+
+The NodeApp in the example has three endpoints:
+
+- `GET /ports`: returns the ports that the application can be accessed on
+- `POST /neworder`: creates a new order
+- `GET /order`: queries an order
+  
+The following demonstration will start from creating a cluster and step-by-step introduce the configuration of the environment, installation and configuration of various components, deployment of applications, and so on.
+
+## One-Click Installation Script
+
+We also provide a script for one-click installation and quick experience, eliminating the tedious configuration of the environment and components. You can visit [GitHub](https://github.com/addozhang/flomesh-dapr-demo) to get the script content.
```

Comment:
```suggestion
For this demo, we provide a quick, one-click installation script to avoid the tedious configuration of the environment and components. [Get the script content from GitHub](https://github.com/addozhang/flomesh-dapr-demo).
```